src/sundials/sundials_logger_impl.h

```diff
@@ -32,6 +32,125 @@
 #define SUNDIALS_LOGGING_EXTRA_DEBUG
 #endif
 
+#if SUNDIALS_LOGGING_LEVEL >= SUNDIALS_LOGGING_ERROR
+#define SUNLogError(logger, scope, label, msg_txt, ...)                 \
+  SUNLogger_QueueMsg(logger, SUN_LOGLEVEL_ERROR, scope, label, msg_txt, \
```

Comment:
Is this portable or will some preprocessors produce `SUNLogger_QueueMsg(loger, SUN_LOGLEVER_ERROR, scope, label,)`?

Reply:
I'm not sure I understand what circumstances could cause the preprocessor to produce that. Apparently old MSVC versions had some buggy behavior with `__VA_ARGS__`, but Visual Studio 2015 handles it fine: https://godbolt.org/z/477aKTosE. I'm not aware of any clang or gcc issues.

Reply:
I was thinking of something like this https://godbolt.org/z/z48KWeEPT

Reply:
Every call to `SUNLogError` already has at least 4 arguments so this shouldn't be an issue. If for some reason a developer passed 3, it will raise an error since an argument is missing (which is a good and expected outcome).

Reply:
I guess with `scope` now removed as an argument, subtract those numbers by 1.

Reply:
I see, it's sort of hacking `__VA_ARGS__` to place the last argument. This gets nicer in C23 with the addition of `__VA_OPT__` to deal with the trailing comma problem.